### PR TITLE
Add self-transform to default form associations

### PR DESCRIPTION
**Pull request by bondjimbond**
**JIRA Ticket**: (https://jira.duraspace.org/browse/ISLANDORA-2383)

# What does this Pull Request do?

Adds cleanup self-transform to the Scholar SP default form associations. Provides a cleaner MODS xml datastream.

This is/will be one of many similar updates, as all of the stock forms should receive this association.

# How should this be tested?
* Ingest a test object with lots of blank fields in the form; review the resulting MODS XML and see how it is filled with a horrible mass of empty XML elements
* Check out the branch
* Review the form association (form builder -> enabled associations)
* Ingest a test object with a lot of blank metadata fields and review the resulting MODS XML, see how it's less horrible than before

# Interested parties
@Islandora/7-x-1-x-committers

### CODE REVIEW

**Comment by bryjbrown** [comment]
@bondjimbond You'll need to do some Git magic to add #316 into this PR so that the Travis tests pass. Assuming you still have your own fork of this repo with the `bondjimbond:fix_form` branch, follow these steps:

1. Set `islandora/islandora_scholar` as a remote source (named 'upstream' here): `git remote add upstream https://github.com/islandora/islandora_scholar.git`
2. `git fetch upstream`
3. Make a local copy of `islandora/islandora_scholar`'s 7.x branch with `git checkout -b upstream/7.x`
4. Checkout your fork's copy of the 7.x branch and merge upstream/7.x into it with `git checkout 7.x` and then `git merge upstream/7.x`
5. Checkout this PRs source branch with `git checkout fix_form` and merge the changes in with `git merge 7.x`.
6. Verify that everything went okay and that the 'fix_form' branch has all the stuff from #316 in it. If so, then `git push origin fix_form`. This should trigger a new Travis build on this latest commit, and that Travis build should now pass.

**Comment by bondjimbond** [comment]
@bryjbrown Since I have islandora's repository as my origin, my process was a little simpler... 

1. git checkout fix_form
2. git pull origin 7.x
3. git push

**Comment by bryjbrown** [comment]
@bondjimbond Showoff.

**Comment by DonRichards** [comment]
I completely missed this. Apparently I'm in my own little world. @bryjbrown Do you need assistance with this?

**Review by bryjbrown — APPROVED** [review]

**Comment by bryjbrown** [comment]
I hereby confirm that this is indeed `less horrible than before` @bondjimbond

**Comment by bondjimbond** [comment]
Excellent! There's a dozen more of these that need approval too. 😳

**Comment by bondjimbond** [comment]
@bryjbrown Just a merge prod reminder. Following are all the others Solution Packs with the same change (which I think we agreed last Committers' Call could just be straight up merged since the code is identical):

- https://github.com/Islandora/islandora_solution_pack_entities/pull/133
- https://github.com/Islandora/islandora_solution_pack_web_archive/pull/51
- https://github.com/Islandora/islandora_solution_pack_video/pull/153
- https://github.com/Islandora/islandora_solution_pack_pdf/pull/137
- https://github.com/Islandora/islandora_solution_pack_newspaper/pull/165
- https://github.com/Islandora/islandora_solution_pack_large_image/pull/188
- https://github.com/Islandora/islandora_solution_pack_disk_image/pull/34
- https://github.com/Islandora/islandora_solution_pack_compound/pull/156
- https://github.com/Islandora/islandora_solution_pack_book/pull/207
- https://github.com/Islandora/islandora_solution_pack_image/pull/140
- https://github.com/Islandora/islandora_solution_pack_audio/pull/149

**Comment by bondjimbond** [comment]
@DonRichards Given that this was merged, can we hit the other 11? No need for testing as code is identical.

**Comment by manez** [comment]
@bondjimbond it's happening :+1:

**Comment by bondjimbond** [comment]
Thanks @manez!

**Pull request merged**
